### PR TITLE
[ACR] Update webhook listEvents and registry SKU

### PR DESCRIPTION
**Pull request by djyou**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by balajikris** [comment]
@djyou : I'll run some checks (syntactic and semantic) on this and get back to you. Meanwhile, Could you please help me understand the intent behind the change - why `listEvents` is a `post` and not `get` ?

**Comment by djyou** [comment]
This is for RBAC purpose so that by default people with `read` permission won't be able to call it.

**Comment by balajikris** [comment]
Hi @djyou :

autorest CI flagged some existing violations in this spec. This can be seen here: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/227625847
Please consider addressing them.

~~~
autorest -i .\azure-rest-api-specs\arm-containerregistry\2017-06-01-preview\swagger\containerregistry.json -g none
WARNING: Since operation 'Registries_RegenerateCredential' response has model definition 'array', it should be of the form "*_list*"
        Path: file:///E:/repos/azure-rest-api-specs/arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json#/paths
WARNING: A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable. Operation: 'Registries_Create' Request Model: 'RegistryCreateParameters' Response Model: 'Registry'
        Path: file:///E:/repos/azure-rest-api-specs/arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json#/paths/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/put/operationId
WARNING: A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable. Operation: 'Webhooks_Create' Request Model: 'WebhookCreateParameters' Response Model: 'Webhook'
        Path: file:///E:/repos/azure-rest-api-specs/arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json#/paths/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/webhooks/{webhookName}/put/operationId
WARNING: The child tracked resource, 'replications' with immediate parent 'Registry', must have a list by immediate parent operation.
        Path: file:///E:/repos/azure-rest-api-specs/arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json#/paths/#/definitions/replications
WARNING: The child tracked resource, 'webhooks' with immediate parent 'Registry', must have a list by immediate parent operation.
        Path: file:///E:/repos/azure-rest-api-specs/arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json#/paths/#/definitions/webhooks
WARNING: The tracked resource, 'Replication', must have a list by resource group operation.
        Path: file:///E:/repos/azure-rest-api-specs/arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json#/definitions/ReplicationWARNING: The tracked resource, 'Webhook', must have a list by resource group operation.
        Path: file:///E:/repos/azure-rest-api-specs/arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json#/definitions/Webhook
ERROR: Tracked resource 'Replication' must have patch operation that at least supports the update of tags.
        Path: file:///E:/repos/azure-rest-api-specs/arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json#/definitions/ReplicationWARNING: Booleans are not descriptive and make them hard to use. Instead use string enums with allowed set of values defined.
        Path: file:///E:/repos/azure-rest-api-specs/arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json#/definitions/RegistryNameStatus/properties/nameAvailable
ERROR: Errors found during Swagger validation
~~~

**Comment by balajikris** [comment]
status:
- [ ] linter validation (pending)
- [x] semantic and examples validation
- [x] manual code review
- [ ] merge PR

**Comment by djyou** [comment]
@balajikris We ended up deciding to update registry SKU as part of PR. Please review and merge this PR if they look good to you. For the warnings you pointed out, we have discussed and agreed on them with previous reviewers, but please let me know if you would like to clarify anything. Thanks. /cc @SajayAntony

**Comment by balajikris** [comment]
hey @djyou :
could you please expand on what was agreed upon with other reviewers? I might be out of the loop here. Was the agreement that these errors/warns do not apply to your spec or is it that you have a workitem to update them at a later time?

File: arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json [hunk 1]
```diff
@@ -1028,8 +1028,9 @@
           "description": "The SKU name of the the container registry. Required for registry creation.",
           "enum": [
             "Basic",
-            "Standard",
-            "Premium"
+            "Standard_Small",
+            "Standard_Medium",
+            "Standard_Large"
```
Comment (balajikris) [inline arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json]: Does this mean the service has already changed? If so, how are you handling requests from existing code that will still come in with values, say "premium" ? I'm trying to understand if this is a breaking change or not.

**Comment by djyou** [comment]
```
"message": "Since operation 'Registries_RegenerateCredential' response has model definition 'array', it should be of the form \"*_list*\"",
```
`Registries_RegenerateCredential` is supposed to return the same response as `Registries_ListCredentials`, similar to the behavior of `StorageAccounts_RegenerateKey`.

```
"message": "A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable. Operation: 'Registries_Create' Request Model: 'RegistryCreateParameters' Response Model: 'Registry'",
"message": "A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable. Operation: 'Webhooks_Create' Request Model: 'WebhookCreateParameters' Response Model: 'Webhook'",
```
The message already explained. It is because "If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable."

```
"message": "The child tracked resource, 'replications' with immediate parent 'Registry', must have a list by immediate parent operation.",
"message": "The child tracked resource, 'webhooks' with immediate parent 'Registry', must have a list by immediate parent operation.",
```
There is a List operation for replications and webhooks to list them as nested resources under a registry.

```
"message": "The tracked resource, 'Replication', must have a list by resource group operation.",
"message": "The tracked resource, 'Webhook', must have a list by resource group operation.",
```
They are nested resources and list by resource group doesn't apply.

```
"message": "Tracked resource 'Replication' must have patch operation that at least supports the update of tags.",
```
PUT is CreateOrUpdate.

```
"message": "Booleans are not descriptive and make them hard to use. Instead use string enums with allowed set of values defined.",
```
These are by design and for backward compatibility.

If possible, can we please limit the scope of this PR to what we are updating?

File: arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json (hunk 1 above)
Comment (djyou) [inline arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json]: This api-version is still under private preview. While this is a breaking change there will be no impact.

**Comment by balajikris** [comment]
While I accept your answers, I am not authorized to do this: _"If possible, can we please limit the scope of this PR to what we are updating?"_ . 
We're supposed to look at "all" the errors/warns in the spec and go through them and arrive at an acceptable resolution, until the spec is clean. In this case, I was simply missing the context and you provided it above. cc: @salameer

File: arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json (hunk 1 above)
Comment (balajikris) [inline arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json]: cool - thanks for clarifying.

**Review by balajikris — APPROVED** [review]

**Pull request merged**